src/app/core/common/hide-overflow-list.directive.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Directive,
+  ElementRef,
+  Input, OnDestroy, OnInit,
+  Renderer2
+} from '@angular/core';
+import { Subscription, timer } from 'rxjs';
+import { distinctUntilChanged, map } from 'rxjs/operators';
+
+@Directive({
+  selector: '[algHideOverflowList]',
```

Comment:
Why using a directive? Wouldnt be easier/cleaner in the component directly?

Reply:
The directive is thing for manipulate HTML structure, it could be done in component it could be done even in index.html but this is not correct.